src/v1.rs

```diff
@@ -149,12 +149,14 @@ impl Party {
         self.private_key = Scalar::zero();
         self.group_key = Point::zero();
 
+        let threshold: usize = self.threshold.try_into().unwrap();
         let mut bad_ids = Vec::new(); //: Vec<u32> = polys
         for (i, comm) in polys.iter() {
-            if !comm.verify() {
+            if comm.poly.len() != threshold || !comm.verify() {
                 bad_ids.push(*i);
+            } else {
+                self.group_key += comm.poly[0];
```

Comment:
I know this was here before, but can `poly` this ever be empty?

Reply:
It certainly can't if it passes the above test.   An old PR from @hstove that's still around discusses how to handle a threshold of `0` (#55).

Reply:
I think we should just `panic` on a `threshold` of `0`.